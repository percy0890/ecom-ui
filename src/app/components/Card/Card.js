import React from 'react';
import PropTypes from 'prop-types';
import IMG from 'app/utils/images';
import AppConstants from 'app/app.constants.json';
import './Card.scss';
import { naCheck } from 'app/utils/common';

class Card extends React.PureComponent {
  clickCard = () => {
    this.props.onCardExpand(this.props.id);
  };

  checkValidationOnSelection = () =>
    this.props.selectedCard &&
    this.props.selectedCard.includes(this.props.id) &&
    this.props.tapType === AppConstants.CLICK_TYPE.SELECT;

  getSelectedClass = () => {
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
    )
      return 'select-card-pickup';
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGOUT
    )
      return 'select-card-drop';
    return '';
  };

  getShiftClass = () => {
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
    )
      return 'shifts-pickup';
    if (
      this.checkValidationOnSelection() &&
      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGOUT
    )
      return 'shifts-drop';
    return '';
  };

  render() {
    return (
      <React.Fragment>
        {this.props.selectedCard &&
        this.props.selectedCard.includes(this.props.id) &&
        this.props.tapType === AppConstants.CLICK_TYPE.EXPAND ? (
          <div
            className="card-container-open"
            role="presentation"
            onClick={this.clickCard}
          >
            <div className={`upper-container ${this.props.customClass}`}>
              <div className="shift-type-block">
                {this.props.shiftIcon && (
                  <img
                    src={this.props.shiftIcon}
                    alt="dots"
                    className="image"
                  />
                )}
                <div className="shift-type">
                  {naCheck(
                    this.props.shiftType && this.props.shiftType.toUpperCase(),
                  )}
                </div>
              </div>
              <div className="shift-id-container">
                <div className="shift-id">
                  <span>SHIFT ID: </span>
                  <span className="text">{this.props.shiftId}</span>
                </div>
                {this.props.assignmentIcon && (
                  <img
                    src={this.props.assignmentIcon}
                    alt="dots"
                    className="image"
                  />
                )}
              </div>
            </div>
            <div className="bottom-container">
              <div className="booking-id">
                <span>BOOKING ID: </span>
                <span className="text">{naCheck(this.props.bookingId)}</span>
              </div>
              <div className="location-container">
                <div className="row-wrap">
                  <div className="date-time">
                    <div>{naCheck(this.props.startDate)}</div>
                    <div className="time">{naCheck(this.props.startTime)}</div>
                  </div>
                  <div className="address-box">
                    <div className="indicator green" />
                    <div className="text">
                      <span>{naCheck(this.props.startLocation)}</span>
                    </div>
                  </div>
                </div>
                <div className="row-wrap">
                  <div className="date-time">
                    <div>{naCheck(this.props.endDate)}</div>
                    <div className="time">{naCheck(this.props.endTime)}</div>
                  </div>
                  <div className="address-box">
                    <div className="indicator red" />
                    <div className="text">
                      <span>{naCheck(this.props.endLocation)}</span>
                    </div>
                  </div>
                </div>
              </div>
              <div className="info-container">
                <div className="info-row">
                  <div className="info-field">
                    <div className="label">Trip Status</div>
                    <div className="value">
                      {naCheck(this.props.tripStatus)}
                    </div>
                  </div>
                  <div className="info-field">
                    <div className="label">Vehicle No.</div>
                    <div className="value">
                      {naCheck(this.props.vehicleNumber)}
                    </div>
                  </div>
                </div>
                <div className="info-row">
                  <div className="info-field">
                    <div className="label">Driver Name</div>
                    <div className="value">
                      {naCheck(this.props.driverName)}
                    </div>
                  </div>
                  <div className="info-field">
                    <div className="label">Contact No.</div>
                    <div className="value">
                      {naCheck(this.props.contactNumber)}
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        ) : (
          <div
            className="card-container cursor-pointer"
            role="presentation"
            onClick={this.clickCard}
          >
            {this.props.selectedCard &&
            this.props.selectedCard.includes(this.props.id) &&
            this.props.tapType === AppConstants.CLICK_TYPE.SELECT ? (
              <div className={`left-container ${this.props.customClass}`}>
                <div className="title-block center-align">
                  <img
                    src={IMG.FILTER_RIDE}
                    alt="dots"
                    className="checked-image"
                  />
                </div>
              </div>
            ) : (
              <div className={`left-container ${this.props.customClass}`}>
                <div className="title-block">
                  {this.props.shiftIcon && (
                    <img
                      src={this.props.shiftIcon}
                      alt="dots"
                      className="image"
                    />
                  )}
                  <div className="title">
                    {naCheck(
                      this.props.shiftType &&
                        this.props.shiftType.toUpperCase(),
                    )}
                  </div>
                </div>
                <div className="description-block">
                  <div>{naCheck(this.props.startDate)}</div>
                  <div className="time">
                    {naCheck(this.props.startTime)}-
                    {naCheck(this.props.endTime)}
                  </div>
                </div>
              </div>
            )}
            <div className={`right-container ${this.getSelectedClass()} `}>
              <div className={`upper-right-container ${this.getShiftClass()} `}>
                <div className="shift-id-block">
                  <span>SHIFT ID: </span>
                  <span className="shift-id">
                    {naCheck(this.props.shiftId)}
                  </span>
                </div>
                {this.props.assignmentIcon && (
                  <img
                    src={this.props.assignmentIcon}
                    alt="dots"
                    className="shift-icon"
                  />
                )}
              </div>
              <div className="address-container">
                <div className="address-indicator-block">
                  {this.props.icon && <img src={this.props.icon} alt="dots" />}
                </div>
                <div className="address">
                  <span>
                    {naCheck(
                      this.props.shiftType === AppConstants.SHIFT_TYPE.LOGIN
                        ? this.props.startLocation
                        : this.props.endLocation,
                    )}
                  </span>
                </div>
              </div>
            </div>
          </div>
        )}
      </React.Fragment>
    );
  }
}

Card.propTypes = {
  id: PropTypes.any,
  customClass: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  startTime: PropTypes.string,
  endTime: PropTypes.string,
  shiftId: PropTypes.string,
  shiftType: PropTypes.string,
  icon: PropTypes.any,
  assignmentIcon: PropTypes.any,
  shiftIcon: PropTypes.any,
  onCardExpand: PropTypes.func,
  selectedCard: PropTypes.any,
  bookingId: PropTypes.string,
  contactNumber: PropTypes.string,
  driverName: PropTypes.string,
  vehicleNumber: PropTypes.string,
  tripStatus: PropTypes.string,
  startLocation: PropTypes.string,
  endLocation: PropTypes.string,
  tapType: PropTypes.string,
};

export default Card;
