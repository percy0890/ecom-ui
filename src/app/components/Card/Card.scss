@import '~app/styles/core';

.card-container {
  display: flex;
  background-color: $app-trip-card-bg;
  width: 46%;
  box-shadow: $app-shadow-container;
  border-radius: 0.5rem;
  margin: 2rem;

  .left-container {
    flex: 1;
    display: flex;
    flex-direction: column;
    color: $app-trip-card-bg;
    border-top-left-radius: inherit;
    border-bottom-left-radius: inherit;
    padding: 2.5rem 2rem;

    .title-block {
      display: flex;
      font-family: $app-font-bold;
      align-items: center;
      padding: 0.3rem 0;
      font-size: 1.4rem;

      &.center-align {
        justify-content: center;
      }

      .checked-image {
        width: 7rem;
      }

      .title {
        margin-left: 0.6rem;
      }

      .image {
        width: 2rem;
      }
    }

    .description-block {
      margin-top: 1.5rem;

      .time {
        font-family: $app-font-bold;
      }
    }

    &.logout-container {
      background-color: $app-logout-color;
    }

    &.login-container {
      background-color: $app-login-color;
    }
  }

  .right-container {
    flex: 3.2;
    display: flex;
    flex-direction: column;

    .shifts-pickup {
      background-color: $app-shift-pickup !important;
      border-radius: 0.8rem;
    }

    .shifts-drop {
      background-color: $app-shift-drop !important;
      border-radius: 0.8rem;
    }

    .upper-right-container {
      margin: 1rem 1.5rem;
      padding: 0.5rem 1rem;
      background-color: $app-container-color;
      display: flex;
      justify-content: space-between;
      border-radius: 0.6rem;

      .shift-id-block {
        font-family: $app-font-medium;

        .shift-id {
          font-family: $app-font-bold;
        }
      }

      .shift-icon {
        width: 2rem;
      }
    }

    .address-container {
      margin: 0 1.5rem 1rem;
      display: flex;

      .address-indicator-block {
        img {
          width: 1.1rem;
        }
      }

      .address {
        padding-left: 0.8rem;
        font-family: $app-font-regular;
      }
    }
  }

  .select-card-pickup {
    background-color: $app-drop-select;
  }

  .select-card-drop {
    background-color: $app-pickup-select;
  }
}

@include media(tablet up) {
  .card-container {
    width: 46%;
    margin: 0.75rem 0.75rem;
  }
}

@include media(mobile down) {
  .card-container {
    width: 100%;
    margin: 0.75rem 0;
  }
}

.card-container-open {
  background-color: $app-trip-card-bg;
  width: 46%;
  box-shadow: $app-shadow-container;
  border-radius: 0.5rem;
  margin: 2rem;

  .upper-container {
    display: flex;
    color: $app-trip-card-bg;
    border-top-left-radius: inherit;
    border-top-right-radius: inherit;
    padding: 1.5rem;
    font-size: 1.6rem;

    &.logout-container {
      background-color: $app-login-color;
    }

    &.login-container {
      background-color: $app-logout-color;
    }

    .shift-type-block {
      flex: 1;
      display: flex;
      font-family: $app-font-bold;
      align-items: center;

      .shift-type {
        flex: 8;
        margin: 0.2rem 1rem;
      }

      .image {
        width: 2rem;
      }
    }

    .shift-id-container {
      flex: 2;
      display: flex;
      align-items: center;
      justify-content: space-between;

      .shift-id {
        padding: 0 1rem;
        font-family: $app-font-regular;

        .text {
          font-family: $app-font-bold;
        }
      }

      .image {
        width: 2rem;
      }
    }
  }

  .bottom-container {
    padding: 1.5rem;
    font-family: $app-font-regular;

    .booking-id {
      font-family: $app-font-regular;
      padding: 1rem 0;

      .text {
        font-family: $app-font-bold;
      }
    }

    .location-container {
      overflow: hidden;

      .row-wrap {
        display: flex;
        margin: 1.5rem 0;
        font-family: $app-font-regular;

        .date-time {
          line-height: 2.2rem;
          flex-shrink: 0;
          padding-right: 2.5rem;

          .time {
            font-family: $app-font-bold;
            font-size: 1.8rem;
          }
        }

        .address-box {
          display: flex;
          flex-grow: 1;

          .indicator {
            width: 1.5rem;
            height: 1.5rem;
            border-radius: 50%;
            flex-shrink: 0;
            position: relative;

            &.green {
              background-color: #7BC141;

              &:after {
                content: '';
                width: 2px;
                border-right: 2px dotted $app-border-color;
                position: absolute;
                top: 1.5rem;
                left: calc(50% - 1px);
                height: 100px;
              }
            }

            &.red {
              background-color: #CE2B04;

              &:after {
                content: '';
                width: 1.5rem;
                background-color: #fff;
                position: absolute;
                top: 1.5rem;
                left: 0;
                height: 50px;
              }
            }
          }
        }

        .text {
          padding-left: 1rem;
          font-size: 1.8rem;
        }
      }
    }

    .info-container {
      .info-row {
        display: flex;
        justify-content: space-between;
        margin-bottom: 1rem;

        .info-field {
          width: 48%;
          line-height: 3rem;

          .label {
            color: $brand-color-primary;
            font-family: $app-font-bold;
          }

          .value {
            padding: 0.5rem 1rem;
            background-color: $app-bg-body;
            border-radius: 0.6rem;
          }
        }
      }
    }
  }
}

@include media(tablet up) {
  .card-container-open {
    width: 46%;
    margin: 0.75rem 0.75rem;
  }
}

@include media(mobile down) {
  .card-container-open {
    width: 100% !important;
    margin: 0.75rem 0rem !important;

    .image {
      width: 2rem !important;
    }

    .dotted-image {
      height: 62.8% !important;
    }
  }
}
