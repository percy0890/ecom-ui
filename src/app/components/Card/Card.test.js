import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = renderer.create(<Card>App</Card>);
  });

  it('renders correctly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Call clickCard function', () => {
    const onCardExpand = jest.fn();
    const props = {
      tapType: 'select',
      shiftType: 'LOGOUT',
      onCardExpand,
    };
    const card = shallow(<Card {...props} />);
    card.instance().clickCard();
    expect(onCardExpand).toBeCalled();
  });

  it('Call getSelectedClass function', () => {
    const onCardExpand = jest.fn();
    const props = {
      tapType: 'SELECT',
      selectedCard: [1],
      id: 1,
      shiftType: 'LOGIN',
      onCardExpand,
    };
    const card = shallow(<Card {...props} />);
    const getSelectedClass = jest.spyOn(card.instance(), 'getSelectedClass');

    expect(getSelectedClass).not.toHaveBeenCalled();
  });
  it('Check Select with logout functionality', () => {
    const onCardExpand = jest.fn();
    const props = {
      tapType: 'SELECT',
      selectedCard: [1],
      id: 1,
      shiftType: 'LOGOUT',
      onCardExpand,
      assignmentIcon: true,
    };
    const card = shallow(<Card {...props} />);
    const getSelectedClass = jest.spyOn(card.instance(), 'getSelectedClass');

    expect(getSelectedClass).not.toHaveBeenCalled();
  });
  it('Check Expand with logout functionality', () => {
    const onCardExpand = jest.fn();
    const props = {
      tapType: 'EXPAND',
      selectedCard: [1],
      id: 1,
      shiftType: 'LOGOUT',
      onCardExpand,
      assignmentIcon: true,
      shiftIcon: true,
    };
    const card = shallow(<Card {...props} />);
    const getSelectedClass = jest.spyOn(card.instance(), 'getSelectedClass');

    expect(getSelectedClass).not.toHaveBeenCalled();
  });
});
