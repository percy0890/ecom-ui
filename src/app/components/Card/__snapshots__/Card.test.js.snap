// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders correctly 1`] = `
<div
  className="card-container cursor-pointer"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="left-container undefined"
  >
    <div
      className="title-block"
    >
      <div
        className="title"
      >
        N/A
      </div>
    </div>
    <div
      className="description-block"
    >
      <div>
        N/A
      </div>
      <div
        className="time"
      >
        N/A
        -
        N/A
      </div>
    </div>
  </div>
  <div
    className="right-container  "
  >
    <div
      className="upper-right-container  "
    >
      <div
        className="shift-id-block"
      >
        <span>
          SHIFT ID: 
        </span>
        <span
          className="shift-id"
        >
          N/A
        </span>
      </div>
    </div>
    <div
      className="address-container"
    >
      <div
        className="address-indicator-block"
      />
      <div
        className="address"
      >
        <span>
          N/A
        </span>
      </div>
    </div>
  </div>
</div>
`;
